pallets/ajuna-board/src/lib.rs

```diff
@@ -243,4 +287,31 @@ impl<T: Config> Pallet<T> {
 			None => Seed::<T>::kill(),
 		}
 	}
+
+	fn declare_winner(
+		board_id: &T::BoardId,
+		winner: &<<T as Config>::Game as TurnBasedGame>::Player,
+		board_state: &<<T as Config>::Game as TurnBasedGame>::State,
+	) {
+		// Cache result in storage, this would be cleared on `flush_winner`
+		BoardWinners::<T>::insert(board_id, winner.clone());
+		BoardExpiries::<T>::remove(board_id);
+		Self::seed_for_next(board_state);
+		Self::deposit_event(Event::GameFinished { board_id: *board_id, winner: winner.clone() });
+	}
+
+	fn set_or_update_expiry(board_id: T::BoardId) {
+		let current_block_number = <frame_system::Pallet<T>>::block_number();
+		let expiry_block_count = T::MaxNumberOfIdleBlocks::get();
+
+		let block_of_expiry = current_block_number + expiry_block_count.into();
+
+		if BoardExpiries::<T>::contains_key(board_id) {
+			BoardExpiries::<T>::mutate(board_id, |entry| {
+				*entry = block_of_expiry;
+			});
+		} else {
+			BoardExpiries::<T>::insert(board_id, block_of_expiry);
+		}
```

Comment:
Nice checks! But these lines can simply be:
```suggestion
		BoardExpiries::<T>::insert(board_id, block_of_expiry);
```
(`insert()` will mutate / replace the key with a new value)

Reply:
Oh! I didn't know that, the docs unfortunately didn't mention such behavior, good to know!